spark/src/main/java/org/opensearch/sql/spark/dispatcher/SparkQueryDispatcher.java

```diff
@@ -106,4 +129,80 @@ private String constructSparkParameters(String datasourceName) throws URISyntaxE
         "spark.sql.catalog." + datasourceName, FLINT_DELEGATE_CATALOG);
     return s3GlueSparkSubmitParameters.toString();
   }
+
+  private StartJobRequest getStartJobRequestForNonIndexQueries(
+      DispatchQueryRequest dispatchQueryRequest) {
+    StartJobRequest startJobRequest;
+    FullyQualifiedTableName fullyQualifiedTableName =
+        SQLQueryUtils.extractFullyQualifiedTableName(dispatchQueryRequest.getQuery());
+    if (fullyQualifiedTableName == null || fullyQualifiedTableName.getDatasourceName() == null) {
+      throw new UnsupportedOperationException("Queries without a datasource are not supported");
+    }
+    dataSourceUserAuthorizationHelper.authorizeDataSource(
+        this.dataSourceService.getRawDataSourceMetadata(
+            fullyQualifiedTableName.getDatasourceName()));
+    String jobName =
+        dispatchQueryRequest.getClusterName()
+            + ":"
+            + fullyQualifiedTableName.getFullyQualifiedName();
+    Map<String, String> tags =
+        getDefaultTagsForJobSubmission(dispatchQueryRequest, fullyQualifiedTableName);
+    startJobRequest =
+        new StartJobRequest(
+            dispatchQueryRequest.getQuery(),
+            jobName,
+            dispatchQueryRequest.getApplicationId(),
+            dispatchQueryRequest.getExecutionRoleARN(),
+            constructSparkParameters(fullyQualifiedTableName.getDatasourceName()),
+            tags);
+    return startJobRequest;
+  }
+
+  private StartJobRequest getStartJobRequestForIndexRequest(
```

Comment:
Only Create Index is handled in this function, and submit SQL to EMR-S. But DROP INDX, DESCRIBE INDEX will not. right?

Reply:
DROP INDEX will be handled in next PR. Will refactor this class then. 
DESCRIBE INDEX is also job submission right? @penghuo